tests/selenium/test_selenium_chat_flow.py

```
@@ -369,3 +369,238 @@ def test_explain_denial_flow_end_to_end(self):
+
+        # User info should still be there
+        user_info_json = self.execute_script(
+            "return localStorage.getItem('fhi_user_info');"
+        )
+        assert user_info_json is not None
+
+        user_info = json.loads(user_info_json)
+        assert user_info["firstName"] == "Existing"
+
+        print("✓ Existing user successfully submitted explain denial")
+        print(f"  - User info preserved from previous session")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)